spec/unit/http/http_client_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'rspec'
+
+module LibyuiClient
+  module Http
+    RSpec.describe HttpClient do
+      let(:http) { Net::HTTPSuccess.new('1.1', 200, 'OK') }
+      before { allow(http).to receive(:body).and_return('{"id": "foo"}') }
+      describe '#http_get' do
+        context 'makes a GET request' do
+          it 'returns a new LibyuiClient::Http::Response' do
+            expect_any_instance_of(Net::HTTP).to receive(:request).with(an_instance_of(Net::HTTP::Get))
+                                                                  .and_return(http)
+            escapedurl = URI.escape('http://test.org')
+            uri = URI.parse(escapedurl)
+            response = HttpClient.http_get(uri)
+            expect(response.code).to eq(200)
+            expect(response.body).to eq('{"id": "foo"}')
+          end
+        end
+      end
+      # of #http_get
+
+      describe '#http_post' do
+        context 'makes a POST request' do
+          it 'returns a new LibyuiClient::Http::Response' do
+            expect_any_instance_of(Net::HTTP).to receive(:request).with(an_instance_of(Net::HTTP::Post))
```

Comment:
here we might want to test the other objects that are created in the method so we could be more specific, instead of expect any instance, but I was not able to give you some running example :( Basically what I wanted to achieve is to expect new method on those object. Another thing is if makes sense or not to do it?

Reply:
i dont remember if i mentioned this but in general we should avoid the use of any_instance https://relishapp.com/rspec/rspec-mocks/v/3-9/docs/working-with-legacy-code/any-instance.  However i cannot tell what you want to achieve exactly. As i said also there are other way too and we should improve this part in the future. But this is kinda advance for me to try to play with at the moment. So yes it makes sense to refactor it but i am not sure if this has to be in this PR